### PR TITLE
[ffmpeg] enable opengl feature on osx

### PR DESCRIPTION
**Pull request by mcmtroffaes**
- #### What does your PR fix?  
  The opengl feature was disabled on osx when compiling ffmpeg[all], however ffmpeg supports it perfectly on osx and it compiles with no problems (see for instance https://github.com/mcmtroffaes/ffmpeg-msvc-build/runs/2903919175).

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  No changes besides adding an extra feature for osx.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes.

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes.

### CODE REVIEW

**Comment by NancyLi1013** [comment]
Hi @mcmtroffaes 

Thanks again for your PR.
```
Building package ffmpeg[all,all-gpl,all-nonfree,ass,avcodec,avdevice,avfilter,avformat,avresample,bzip2,core,fdk-aac,fontconfig,freetype,fribidi,gpl,iconv,ilbc,lzma,modplug,mp3lame,nonfree,opencl,opengl,openh264,openjpeg,openssl,opus,postproc,snappy,soxr,speex,swresample,swscale,theora,vorbis,vpx,webp,x264,x265,zlib]:x64-osx...
CMake Error at ports/ffmpeg/portfile.cmake:76 (message):
  Feature 'opengl' does not support 'uwp | (windows & arm) | osx'
Call Stack (most recent call first):
  scripts/ports.cmake:139 (include)
```
Seems this is still needed to update.

https://github.com/microsoft/vcpkg/blob/6e024e744e7717c06ddacd5089401109c6298553/ports/ffmpeg/portfile.cmake#L75-L76

**Comment by mcmtroffaes** [comment]
Oh yes, I missed it as I removed these checks in my CI to allow testing the more experimental features. Fixed now.

File: ports/ffmpeg/portfile.cmake [hunk 1]
```diff
@@ -72,7 +72,7 @@ if("opencl" IN_LIST FEATURES)
 endif()
 
 if("opengl" IN_LIST FEATURES)
-    if (((VCPKG_TARGET_ARCHITECTURE STREQUAL "arm" OR VCPKG_TARGET_ARCHITECTURE STREQUAL "arm64") AND VCPKG_TARGET_IS_WINDOWS) OR VCPKG_TARGET_IS_UWP OR VCPKG_TARGET_IS_OSX)
+    if (((VCPKG_TARGET_ARCHITECTURE STREQUAL "arm" OR VCPKG_TARGET_ARCHITECTURE STREQUAL "arm64") AND VCPKG_TARGET_IS_WINDOWS) OR VCPKG_TARGET_IS_UWP)
         message(FATAL_ERROR "Feature 'opengl' does not support 'uwp | (windows & arm) | osx'")
```
Comment (NancyLi1013) [inline ports/ffmpeg/portfile.cmake:76]:
```suggestion
        message(FATAL_ERROR "Feature 'opengl' does not support 'uwp | (windows & arm)")
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
You should have tested this feature `opengl `on osx, right?

**Comment by mcmtroffaes** [comment]
> You should have tested this feature `opengl `on osx, right?

Well, I don't have a mac myself for in-depth testing, however I've run a simple test application on CI https://github.com/mcmtroffaes/ffmpeg-msvc-build/runs/2903919175#step:14:1122 which shows that the opengl device can be opened on osx.

**Comment by NancyLi1013** [comment]
The feature `opengl `has passed on x64-osx on my local machine.

```
Building package ffmpeg[avcodec,avdevice,avfilter,avformat,core,gpl,opengl,postproc,swresample,swscale]:x64-osx... done
Installing package ffmpeg[avcodec,avdevice,avfilter,avformat,core,gpl,opengl,postproc,swresample,swscale]:x64-osx...
Installing package ffmpeg[avcodec,avdevice,avfilter,avformat,core,gpl,opengl,postproc,swresample,swscale]:x64-osx... done
Elapsed time for package ffmpeg:x64-osx: 5.426 min
```
Thanks again for your efforts @mcmtroffaes.

**Comment by mcmtroffaes** [comment]
You're welcome, and thanks for running an extra local test. I'm pretty confident that if it works on CI it should also work for users but better be safe than sorry!

**Comment by mcmtroffaes** [comment]
Merged master & bumped port version again.

**Comment by BillyONeal** [comment]
Thanks for your contribution!

**Pull request merged**